<launch>
    <arg name="output" default="screen"/>
    <arg name="imu" default="imu"/>

    <arg name="port" default="/dev/ttyUSB0" />
    <arg name="frame_id" default="$(arg imu)"/>
    <arg name="baudrate" default="921600"/>
    <arg name="imu_rate" default="100"/>

    <!-- Sync out settings -->
    <!-- to disable this just set it to < 0 -->
    <arg name="sync_rate" default="0"/>
    <arg name="sync_pulse_width_us" default="1000"/>

    <!-- Binary settings -->
    <arg name="binary_output" default="true"/>
    <arg name="binary_async_mode" default="2"/>

    <arg name="compensated" default="false"/>

    <!-- VPE settings -->
    <arg name="vpe_enable" default="true"/>
    <arg name="vpe_heading_mode" default="1"/>
    <arg name="vpe_filtering_mode" default="1"/>
    <arg name="vpe_tuning_mode" default="1"/>

    <!-- Ros Topic settings -->
    <arg name="enable_mag_pres" default="true"/>
    <arg name="enable_ypr" default="false"/>

    <node pkg="imu_vn_100" name="$(arg imu)" type="imu_vn_100" output="$(arg output)">
        <param name="port" type="string" value="$(arg port)"/>
        <param name="baudrate" type="int" value="$(arg baudrate)"/>
        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="imu_rate" type="int" value="$(arg imu_rate)"/>
        <param name="binary_output" type="bool" value="$(arg binary_output)"/>
        <param name="binary_async_mode" type="int" value="$(arg binary_async_mode)"/>
        <param name="compensated" type="bool" value="$(arg compensated)"/>

        <param name="enable_mag_pres" type="bool" value="$(arg enable_mag_pres)"/>
        <param name="enable_ypr" type="bool" value="$(arg enable_ypr)"/>

        <param name="sync_rate" type="int" value="$(arg sync_rate)"/>
        <param name="sync_pulse_width_us" type="int" value="$(arg sync_pulse_width_us)"/>

        <param name="vpe/enable" type="bool" value="$(arg vpe_enable)"/>
        <param name="vpe/heading_mode" type="int" value="$(arg vpe_heading_mode)"/>
        <param name="vpe/filtering_mode" type="int" value="$(arg vpe_filtering_mode)"/>
        <param name="vpe/tuning_mode" type="int" value="$(arg vpe_tuning_mode)"/>
    </node>
</launch>
